src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -218,6 +224,63 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 			this._lastUserData = e.key;
 			this._lastUserDataTimestamp = Date.now();
 		}));
+		this._register(this._terminal.onData(e => {
+			if (e === '\x03') {
+				// Ctrl+C
+				this._sendTelemetryInfo(true);
+				this._mostRecentAcceptedCompletionsForCurrentExecution = undefined;
+			}
+		}));
+	}
+
+	private _sendTelemetryInfo(fromInterrupt?: boolean, exitCode?: number): void {
+		const commandLine = this._mostRecentPromptInputState?.value;
+		for (const completion of this._mostRecentAcceptedCompletionsForCurrentExecution || []) {
+			const label = completion?.label;
+			const kind = completion?.kindLabel;
```

Comment:
Instead of peppering this stuff around this file it could all live in a terminalSuggestTelemetry file with a method like `acceptCompletion`, the other listeners might all work by passing in the commanddetection?